lib/srv/regular/sshserver.go

```diff
@@ -338,6 +338,17 @@ func SetProxyMode(tsrv reversetunnel.Server) ServerOption {
 func SetLabels(labels map[string]string,
 	cmdLabels services.CommandLabels) ServerOption {
 	return func(s *Server) error {
+		for key := range labels {
+			if !services.IsValidLabelKey(key) {
+				return trace.BadParameter("invalid label key: %q", key)
+			}
+		}
+
+		for key := range cmdLabels {
+			if !services.IsValidLabelKey(key) {
+				return trace.BadParameter("invalid label key: %q", key)
+			}
+		}
 		// make sure to clone labels to avoid
 		// concurrent writes to the map during reloads
 		cmdLabels = cmdLabels.Clone()
```

Comment:
Combine this with the loop below.
Also, should `labels` be cloned to prevent concurrent writes?

Reply:
As far as I can tell, only `cmdLabels` is mutated.  The `labels` value should be calculated once at program start based on the config file and/or command-line inputs.